app/grandchallenge/algorithms/templates/algorithms/result_list.html

```diff
@@ -83,14 +83,16 @@ <h2>Results for {{ algorithm.title }}</h2>
                     </td>
                     <td>
                         <ul class="list-unstyled">
-                            {% for image in result.images.all %}
+                            {% if result.images %}
                                 <li>
-                                    <a href="{% url 'workstations:workstation-session-redirect' slug=algorithm.workstation.slug %}?{% workstation_query image=result.job.image overlay=image config=algorithm.workstation_config %}">
-                                            <span class="badge badge-primary">
-                                                <i class="fa fa-eye"></i> View Input &amp; Output
-                                            </span>
+                                    <a href="{% url 'workstations:workstation-session-redirect' slug=algorithm.workstation.slug %}?{% workstation_query algorithm_result=result config=algorithm.workstation_config %}">
+                                        <span class="badge badge-primary">
+                                            <i class="fa fa-eye"></i> View Result
+                                        </span>
                                     </a>
                                 </li>
+                            {% endif %}
+                            {% for image in result.images.all %}
```

Comment:
On line 113, currently: 
```
<a href="{% url 'workstations:workstation-session-redirect' slug=algorithm.workstation.slug %}?{% workstation_query image=result.job.image config=algorithm.workstation_config %}
```

This should be updated to use the result API too right?

Reply:
(even if there a no output images produced, maybe the metrics json is still interesting to view next to the image)

Reply:
It's a bit strange as there's always at least one image expected. But can change that so CirrusCore can handle it.